packages/core/saved-objects/core-saved-objects-migration-server-internal/src/zdt/model/model.test.ts

```diff
@@ -113,22 +114,39 @@ describe('model', () => {
   });
 
   describe('dispatching to correct stage', () => {
-    test('dispatching INIT state', () => {
-      const state: State = {
+    const createStubState = (controlState: AllActionStates): State =>
+      ({
         ...baseState,
-        controlState: 'INIT',
-      };
-      const res: StateActionResponse<'INIT'> = Either.right({
+        controlState,
+      } as unknown as State);
+
+    const createStubResponse = (): ResponseType<AllActionStates> =>
+      Either.right({
         '.kibana_7.11.0_001': {
           aliases: {},
           mappings: { properties: {} },
           settings: {},
         },
-      });
-      model(state, res, context);
+      }) as StateActionResponse<'INIT'>;
```

Comment:
NIT feels weird to use the Response type of the INIT stage for each test.
Is it mandatory to cast it here? It seems that it is cast inside the `model()` anyway.